sled-agent/src/services.rs

```diff
@@ -2403,33 +2398,13 @@ impl ServiceManager {
             return;
         }
 
-        let now = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("SystemTime before UNIX EPOCH");
-
-        info!(self.inner.log, "Setting boot time to {:?}", now);
-
-        let files: Vec<Utf8PathBuf> = zones
-            .map(|z| z.root())
-            .chain(iter::once(Utf8PathBuf::from("/")))
-            .flat_map(|r| [r.join("var/adm/utmpx"), r.join("var/adm/wtmpx")])
-            .collect();
-
-        for file in files {
-            let mut command = std::process::Command::new(PFEXEC);
-            let cmd = command.args(&[
-                "/usr/platform/oxide/bin/tmpx",
-                &format!("{}", now.as_secs()),
-                &file.as_str(),
-            ]);
-            match execute(cmd) {
-                Err(e) => {
-                    warn!(self.inner.log, "Updating {} failed: {}", &file, e);
-                }
-                Ok(_) => {
-                    info!(self.inner.log, "Updated {}", &file);
-                }
-            }
+        // Call out to the 'tmpx' utility program which will rewrite the wtmpx
+        // and utmpx databases in every zone, including the global zone, to
+        // reflect the adjusted system boot time.
+        let mut command = std::process::Command::new(PFEXEC);
+        let cmd = command.args(&["/usr/platform/oxide/bin/tmpx", "-Z"]);
+        if let Err(e) = execute(cmd) {
+            warn!(self.inner.log, "Updating [wu]tmpx databases failed: {}", e);
```

Comment:
We lost the `info!` log if this succeeds (although maybe it's less interesting to log "we successfully told `tmpx` to do its thing" than "we told tmpx to modify a specific file"?). I have no strong feeling either way but just wanted to confirm this is intentional.

Reply:
Yes, I had the same thought. The new `tmpx` is fire and forget, it will traverse all zones and idempotently fix up boot times, producing an error if anything goes wrong. I didn't think it was valuable to keep the success line (but I also have no strong feeling)